.github/workflows/cypress-ete-okta.yml

```diff
@@ -56,10 +56,28 @@ jobs:
           name: build
       - name: Unzip build
         run: unzip -q build.zip
+      - uses: pnpm/action-setup@v3
+        with:
+          version: latest
       - uses: actions/setup-node@v4
         with:
           node-version-file: '.nvmrc'
-          cache: 'yarn'
+      - name: Get pnpm cache directory path
+        id: pnpm-cache-dir
+        run: echo "PNPM_STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
+      - name: Get cypress cache directory path
+        id: cypress-cache-dir
+        run: echo "CYPRESS_BIN_PATH=$(pnpm dlx cypress cache path)" >> $GITHUB_OUTPUT
+      - name: Cache pnpm modules and cypress binary
+        uses: actions/cache@v4
+        id: pnpm-cache
+        with:
+          path: |
+            ${{ steps.pnpm-cache-dir.outputs.PNPM_STORE_PATH }}
+            ${{ steps.cypress-cache-dir.outputs.CYPRESS_BIN_PATH }}
+          key: ${{ runner.os }}-pnpm-cypress-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-cypress-
```

Comment:
this crazy stuff lets us cache the modules and cypress binaries so they can be shared between all of the different workflows and machines